src/middleware.js

```diff
@@ -0,0 +1,25 @@
+import { NextResponse } from "next/server";
+import { session } from "./lib/server-session";
+import * as jose from "jose";
+
+export async function middleware(request) {
+  const { token } = session();
+  const secretKey = process.env.SECRET_KEY;
+  const secret = new TextEncoder().encode(secretKey);
+
+  if (!token) {
+    return NextResponse.redirect(new URL("/login", request.url));
+  }
+
+  //validating token
+  try {
+    await jose.jwtVerify(token, secret);
```

Comment:
Why use this if we already have jwt-decode?

Reply:
@alfonsusac jwt-decode just decode the token, it does not verify if the token valid or not